client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -108,6 +108,17 @@ public void readerOffline(String readerId, Position lastPosition) {
 
     @Override
     public CompletableFuture<Checkpoint> initiateCheckpoint(String checkpointName, ScheduledExecutorService backgroundExecutor) {
+
+        synchronizer.fetchUpdates();
+        int maxOutstandingCheckpointRequest = this.synchronizer.getState().getConfig().getMaxOutstandingCheckpointRequest();
+        int currentOutstandingCheckpointRequest = synchronizer.getState().getCheckpointState().getUnCheckpointedHostsSize();
+        if (currentOutstandingCheckpointRequest == maxOutstandingCheckpointRequest) {
```

Comment:
Please use `if (currentOutstandingCheckpointRequest >= maxOutstandingCheckpointRequest)` because it more clearly expresses the intent and is more robust if the RG is reconfigured (which I believe is possible, not 100% sure).